static/app/components/organizations/pageFilters/container.tsx

```diff
@@ -150,21 +148,6 @@ function Container({
       return;
     }
 
-    // We may need to re-initialize the URL state if we completely clear
-    // out the global selection URL state, for example by navigating with
-    // the sidebar on the same view.
-    const oldSelectionQuery = extractSelectionParameters(lastQuery.current);
-    const newSelectionQuery = extractSelectionParameters(location.query);
-
-    // XXX: This re-initialization is only required in new-page-filter
-    // land, since we have implicit pinning in the old land which will
-    // cause page filters to commonly reset.
-    if (isEmpty(newSelectionQuery) && !isEqual(oldSelectionQuery, newSelectionQuery)) {
-      doInitialization();
-      lastQuery.current = location.query;
-      return;
-    }
-
```

Comment:
Removing this to fix a bug: when a new user with no page filter value in local storage selects an environment (say `prod`) and then clicks the "Reset" button, the environment filter will be cleared but then reverted back to `prod`. This happens due to the extraneous `doInitialization()` call written here. It was implemented as a bug fix for a different problem, but we don't seem to need it anymore since we're reverting back to the old persistence behavior.